Ae 497 add funder manually

andrewebdev opened the pull request:
## Description

- Added the missing page for adding a funder manually.
- Added new routes for this page.
- Added new graphql mutations related to this page.

Fixes #497

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Added unit tests to test that we get the expected behaviour.
- Manual testing by running through the create-project workflow, and choosing "Add Funder Manually" button during the search for funder step. Entered the details and checked that the funder is created and we are redirected to the project funding page.

## Checklist:
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [x] I updated the CHANGELOG.md and added documentation if necessary
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have completed manual or automated accessibility testing for my changes
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

jupiter007 commented on app/[locale]/projects/[projectId]/fundings/add/page.tsx:
```diff
@@ -0,0 +1,306 @@
+      <div ref={topRef} />
+      <PageHeader
+        title={editFunding('title')}
+        description={editFunding('description')}
```
L 221: The title should probably be "Add Funding"

jupiter007 commented on app/[locale]/projects/[projectId]/fundings/add/page.tsx:
```diff
@@ -0,0 +1,306 @@
+
+            <FormSelect
+              label={editFunding('labels.fundingStatus')}
+              isRequired
```
The only required field should be `Funder name` but I'm currently seeing all fields required: 
<img width="744" height="714" alt="image" src="https://github.com/user-attachments/assets/9bdd5b73-abf6-46dd-9e4b-8808379348f4" />

andrewebdev replied:
Okay, only the name field is now required.

jupiter007 commented on app/[locale]/projects/[projectId]/fundings/add/page.tsx:
```diff
@@ -0,0 +1,306 @@
+        <ContentContainer>
+          <ErrorMessages errors={errors} ref={errorRef} />
+          <Form onSubmit={handleFormSubmit}>
+            <FormInput
```
I checked with Brian, and he said that this "Funder name" field should be a typeahead search.

jupiter007 commented on app/[locale]/projects/[projectId]/fundings/add/page.tsx:
```diff
@@ -0,0 +1,306 @@
+              isInvalid={(!fundingData.funderOpportunityNumber || !!fieldErrors.funderOpportunityNumber)}
+              errorMessage={fieldErrors.funderOpportunityNumber.length > 0 ? fieldErrors.funderOpportunityNumber : editFunding('messages.errors.fundingProjectNumber')}
+            />
+
```
I filled out all the fields, but I still got "Invalid field" errors and couldn't submit:
<img width="902" height="707" alt="image" src="https://github.com/user-attachments/assets/6a913c8c-303e-40e5-a021-01aca7dcf2a2" />

andrewebdev commented:
Okay @jupiter007 I've addressed your concerns. Could you check again? I also realized that I didn't hook up the project search page on the funder edit section. So I quickly added this as well.

jupiter007 reviewed (approved):
Thanks for making the updates. LGTM
